ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,389 @@
+
+
+##### *upstream-fail-timeout:*
+
+Sets the time during which the specified number of unsuccessful attempts to communicate with the [server](http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream) should happen to consider the server unavailable
+
+
+##### *use-proxy-protocol:*
+
+Enables or disables the use of the [PROXY protocol](https://www.nginx.com/resources/admin-guide/proxy-protocol/) to receive client connection (real IP address) information passed through proxy servers and load balancers such as HAproxy and Amazon Elastic Load Balancer (ELB).
+
+
+##### *use-gzip:*
+
+Enables or disables the use of the nginx module that compresses responses using the ["gzip" module](http://nginx.org/en/docs/http/ngx_http_gzip_module.html)
+
+
+##### *use-http2:*
+
+Enables or disables the [HTTP/2](http://nginx.org/en/docs/http/ngx_http_v2_module.html) support in secure connections 
+
+
+##### *gzip-types:*
+
+MIME types in addition to "text/html" to compress. The special value "*"" matches any MIME type.
+Responses with the "text/html" type are always compressed if `use-gzip` is enabled
+
+
+##### *worker-processes:*
+
+Sets the number of [worker processes](http://nginx.org/en/docs/ngx_core_module.html#worker_processes). By default "auto" means number of available CPU cores
+
+
+### Default configuration options
+
+Running `/nginx-ingress-controller --dump-nginx-configuration` is possible to get the value of the options that can be changed.
+The next table shows the options, the default value and a description
+
+|name                 |default|
+|---------------------------|------|
+|body-size|1m|
+|custom-http-errors|" "|
+|enable-sticky-sessions|"false"|
+|enable-vts-status|"false"|
+|error-log-level|notice|
+|gzip-types||
+|hsts|"true"|
+|hsts-include-subdomains|"true"|
+|hsts-max-age|"15724800"|
+|keep-alive|"75"|
+|max-worker-connections|"16384"|
+|proxy-connect-timeout|"5"|
+|proxy-read-timeout|"60"|
+|proxy-real-ip-cidr|0.0.0.0/0|
+|proxy-send-timeout|"60"|
+|retry-non-idempotent|"false"|
+|server-name-hash-bucket-size|"64"|
+|server-name-hash-max-size|"512"|
+|ssl-buffer-size|4k|
+|ssl-ciphers||
+|ssl-protocols|TLSv1 TLSv1.1 TLSv1.2|
+|ssl-session-cache|"true"|
+|ssl-session-cache-size|10m|
+|ssl-session-tickets|"true"|
+|ssl-session-timeout|10m|
+|use-gzip|"true"|
+|use-http2|"true"|
+|vts-status-zone-size|10m|
+|worker-processes|<number of CPUs>|
+
+
+### Websockets
+
+Support for websockets is configured in NGINX. The only requirement to avoid the close of connections is the increase of the values of `proxy-read-timeout` and `proxy-send-timeout`. The default value of this settings is `30 seconds`. 
+A more adequate value to support websockets is a value higher than one hour (`3600`)
+
+
+### NGINX status page
```

Comment:
I would move this to the main page, they're useful in general and not really about customization